thavalon-server/src/game/state/proposing.rs

```diff
@@ -6,9 +6,43 @@ use super::prelude::*;
 pub struct Proposing {
     /// The player whose turn it is to propose
     proposer: String,
+    selected_players: HashSet<String>,
 }
 
 impl GameState<Proposing> {
+    pub fn handle_player_selected(mut self, player: &str, added_player: String) -> ActionResult {
+        if player != self.phase.proposer {
+            return self.player_error("It's not your proposal");
+        }
+
+        self.phase.selected_players.insert(added_player.clone());
+        let effects = vec![Effect::Broadcast(Message::ProposalUpdated {
+            players: self.phase.selected_players.clone(),
+        })];
+
+        (GameStateWrapper::Proposing(self), effects)
+    }
+
+    pub fn handle_player_unselected(
+        mut self,
+        player: &str,
+        removed_player: String,
+    ) -> ActionResult {
+        if player != self.phase.proposer {
+            return self.player_error("It's not your proposal");
```

Comment:
This string is used twice. Should this be a const somewhere?

Reply:
Sounds good